pyccel/codegen/printing/fcode.py

```diff
@@ -460,32 +460,71 @@ def _print_PythonPrint(self, expr):
         code = ''
         empty_end = FunctionCallArgument(LiteralString(''), 'end')
         space_end = FunctionCallArgument(LiteralString(' '), 'end')
-        kwargs = [f for f in expr.expr if f.has_keyword]
-        for f in kwargs:
-            if f.keyword == 'sep':
-                sep = str(f.value)
-            elif f.keyword == 'end':
-                end = str(f.value)
+        for f in expr.expr:
+            if f.has_keyword:
+                if f.keyword == 'sep':
+                    sep = str(f.value)
+                elif f.keyword == 'end':
+                    end = str(f.value)
         args_format = []
         args = []
         orig_args = [f for f in expr.expr if not f.has_keyword]
 
-        def formatted_args_to_print(args_format, args, end):
-            if args_format == ['*']:
-                return ', '.join(['print *', *args]) + '\n'
-            args_format = ' A '.join(args_format)
-            new_line = "yes" if end.count('\n') > 0 else "no"
-            end = end.replace('\n', '')
-            args_code = ', " " ,'.join([*args])
+        def formatted_args_to_print(fargs_format, fargs, end):
+
+            def split_long_argument(line):
+                line = line[1 : len(line) - 1]
+                length = len(line)
+                longest_chunk = length
+                cuts = []
+                positions = [idx for idx in range(length) if line[idx] == ' ']
+                if len(positions) > 12:
+                    positions = positions[int(len(positions) / 4) : 12]
+                for mask in range(1, 1<<len(positions)):
+                    mask_cuts = [positions[idx] for idx in range(len(positions)) if (1 << idx)&mask != 0 ]
+                    mask_cuts.append(length)
+                    valid = True
+                    mask_longest_chunk = 0
+                    for idx in range(len(mask_cuts) - 1):
+                        if mask_cuts[idx + 1] - mask_cuts[idx] < 8:
+                            valid = False
+                        mask_longest_chunk = max(mask_longest_chunk, mask_cuts[idx + 1] - mask_cuts[idx])
+                    if valid and mask_longest_chunk < longest_chunk:
+                        cuts = mask_cuts
+                        longest_chunk = mask_longest_chunk
+                if longest_chunk != length:
+                    return [line[a + 1 : b + 1] for a, b in zip([-1, *cuts], cuts)]
+                return [line[i:i + int(length / 3 + 1)] for i in range(0, length, int(length / 3 + 1))]
```

Comment:
This function will find the white spaces in a long string ( > 30 characters) then it will try all possible choices of white spaces to choose as a splitting point, out of these it will pick the splitting that minimizes the length of the longest chunk. Do you think we should choose a better heuristic maybe the number of chunks or the average length of all chunks? If no white spaces are present, then the string is evenly cut into 3 parts.

Reply:
So the following:
```
print("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore ")
```
would be translated to:
```
write (*,continue=no) "Lorem ipsum dolor sit amet, consectetur adipiscing elit, "
write (*,continue=yes) "sed do eiusmod tempor incididunt ut labore et dolore "
```
?
This heuristic seems fine to me, if perhaps overly complicated. I would be fine with printing:
```
write (*,continue=no) "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor "
write (*,continue=yes) "incididunt ut labore et dolore "
```
i.e. just truncating at the last acceptable space. Testing all possible combinations is potentially quite costly and I'm not sure that the gain in readability is worth the extra cost

Reply:
If we want the chosen cut to be a space, truncating at the last acceptable space may not work. we could have something like this:
`print("Loremipsumdolorsitamet,consecteturadipiscinelit,seddoeiusmodtemporincididuntutlaboreet dolore ")`
cutting at the last space won't fix the problem. We could apply a blind approach only if we don't value a cut at a space character enough. Also, testing all possible combinations is not costly at all since we only test 2^12 combinations.

Reply:
I am suggesting something like:
```
possible_breakpts = [match.start() for match in re.finditer('[^a-zA-Z0-9]', line)]
acceptable_breakpts = [pt for pr in possible_breakpts  if pt < MAX_SIZE]
if len(acceptable_breakpts)==0:
   # break in word at MAX_SIZE
else:
   # break at acceptable_breakpts[-1]
```

Reply:
Oh i see, i over complicated things. So, I just get rid of all the combinations testing?

Reply:
Yes I think so. Sorry ^^